programs/bpf/rust/simulation/src/lib.rs

```diff
@@ -0,0 +1,41 @@
+use {
+    solana_program::{
+        account_info::{next_account_info, AccountInfo},
+        clock::Clock,
+        declare_id,
+        entrypoint::ProgramResult,
+        msg,
+        pubkey::Pubkey,
+        sysvar::Sysvar,
+    },
+    std::convert::TryInto,
+};
+
+declare_id!("Sim1jD5C35odT8mzctm8BWnjic8xW5xgeb5MbcbErTo");
+
+solana_program::entrypoint!(process_instruction);
+
+pub fn process_instruction(
+    _program_id: &Pubkey,
+    accounts: &[AccountInfo],
+    _instruction_data: &[u8],
+) -> ProgramResult {
+    let account_info_iter = &mut accounts.iter();
+    let slot_account = next_account_info(account_info_iter)?;
+
+    // Slot is an u64 at the end of the structure
+    let data = slot_account.data.borrow();
+    let slot: u64 = u64::from_le_bytes(data[data.len() - 8..].try_into().unwrap());
+
+    let clock = Clock::get().unwrap();
```

Comment:
> Just to confirm only Clock and SlotHistory expose the slot right?

Explicitly, yes.  You could maybe suss it out through the recent blockhashes or slot hashes, but they don't give an explicit slot.

> Do we want to test for SlotHistory::get() as well, to check the other way to obtain the sysvar?

Thankfully, it's not possible to do `SlotHistory::get()`! Which makes this implementation even tidier.  Otherwise we'd need to also override it in the sysvar cache.

> Also for the case that Clock is included in the accounts rather than obtained through Clock::get?

Good call, I'll add that case as well to be sure.  Clock is only updated at the start of the slot, and not at the end, so we should be good, but but it's better to add a test for the future.